File: backend/internal/schemas/category.go
package schemas

type GetAllCategoryDTO struct {
	Type   string `json:"type" binding:"omitempty,oneof=income expense"`
	Take   int64  `json:"take" binding:"required,min=1"`
	Offset int64  `json:"offset" binding:"min=0"`
}
type CreateCategory struct {
	Name  string  `json:"name" binding:"required,min=1,max=100"`
	Type  string  `json:"type" binding:"required,oneof=income expense"`
	Color *string `json:"color" binding:"omitempty,len=7"`
}

File: frontend/lib/queries/category.ts
// lib/queries/transactions.ts (or wherever you have it)
import { ResponseGet } from "@/types/common-request.type";

interface GetCategoryDTO {
    type: "income" | "expense",
    take: number,
    offset: number,
}
interface ResponseGetCategoryData {
    id: string,
    name: string,
    type: "expense" | "income",
    color: string,
    created_at: string
}

export interface ResponseGetCategoryDTO extends ResponseGet<ResponseGetCategoryData> { }

export async function getCatogories(
    {
        take = 10,
        offset = 0,
        type,
    }: GetCategoryDTO
): Promise<ResponseGetCategoryDTO> {
    const params = new URLSearchParams({
        take: take.toString(),
        offset: offset.toString(),
    });

    if (type) {
        params.append('type', type);
    }

    const res = await fetch(`/api/category?${params.toString()}`);
    if (!res.ok) {
        throw new Error('Failed to fetch category');
    }
    return res.json();
}